### PR TITLE
Website: new page for GitHub insights

### PR DESCRIPTION
**Pull request by ssandino**
This PR solves issue #881 by introducing the following changes:

**Phase 1:** 
  1. [x] Creation of a /open-source page:
    - Implements the layout and structure for the new /open-source page.
  2. [x] Addition of an EN translation file:
    - Adds a new English (en) translation file containing text elements specific to the /open-source page.
  3. [x] Add code to retrieve contributors’ info from our GitHub repo, disregarding [rate limits](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28) for unauthenticated requests.
  
 **Phase 2:**
  1. [x] See with @mkue how to best use token to have authorized access. Answer: Personal Access Token (PAT) stored as Vercel secret. This should address both issues: increasing the rate limit overall and resolving the Vercel preview link limit.
  2. [x] Implement new authentication method and test (locally, preview link)
  3. [x] Adding additional information as shown in this [Figma design](https://www.figma.com/design/qGO3YI21AWIjWEyMPGUczM/Social-Income-Main-(Web%2C-App)?node-id=4205-3003&t=GICWytExzOfy6xkj-1). Make different sections and use the shadcn components [cards](https://ui.shadcn.com/docs/components/card) (good [template](https://ui.shadcn.com/examples/dashboard)) and [table](https://ui.shadcn.com/docs/components/table) (good [template](https://ui.shadcn.com/examples/tasks)).

## Release Notes

- **New Features**
  - Introduced a structured overview of open-source contributions, including sections for contributors, issues, and project metrics.
  - Added components for displaying contributor information and filtering issues by labels.
  - Implemented functions to fetch and display metrics such as total forks, commits, stars, and open issues from GitHub.

- **Bug Fixes**
  - Enhanced error handling for API requests to ensure robustness.

- **Documentation**
  - Updated localization strings for the open-source insights webpage.

- **Chores**
  - Added new dependencies and updated existing ones in the project configuration.

### CODE REVIEW

**Comment by ssandino** [comment]
@activus-d new PR for you

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/contributors.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	author: {
+		id: number;
+		login: string;
+		avatar_url: string;
+	};
+	total: number;
+}
+
+export async function getContributors(): Promise<Contributor[]> {
+	const res = await fetch(`https://api.github.com/repos/${owner}/${repo}/stats/contributors`);
```
Comment (mkue) [inline website/src/app/[lang]/[region]/(website)/open-source/(components)/contributors.tsx:21]: Please update this fetch request to use the following header:

```ts
`Authorization: Bearer ${process.env.GITHUB_PAT}`
```

This hopefully fixes the issue that the request didn't work on the deployment branch.

```ts
const res = await fetch(`https://api.github.com/repos/${owner}/${repo}/stats/contributors`, {
  headers: {
    'Authorization': `Bearer ${process.env.GITHUB_PAT}`,
    'Accept': 'application/vnd.github+json'
  }
});
```

Comment (mkue) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/contributors.tsx:21]: To test this locally, you might need to create a token yourself and add it to the .env file.

You can create a token here: https://github.com/settings/tokens?type=beta

Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/contributors.tsx]: Thank you for the help @mkue

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-contributors.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		throw new Error('Expected contributors to be an array, but received a different format.');
+	}
+
+	return contributors
+		.map((contributor: GitHubContributor) => ({
+			id: contributor.author.id,
+			name: contributor.author.login,
+			commits: contributor.total,
+			avatarUrl: contributor.author.avatar_url,
+		}))
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-contributors.tsx:61]: fixed

File: website/src/app/[lang]/[region]/(website)/open-source/(sections)/contributors.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+};
+
+export async function OpenSourceContributors({ lang }: DefaultParams) {
+	const translator = await Translator.getInstance({
+		language: lang,
+		namespaces: ['website-open-source'],
+	});
+
+	const metadata: Metadata = translator.t('metadata');
+	const heading = metadata.heading;
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(sections)/contributors.tsx:17]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(sections)/contributors.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+	const contributors = await getContributors();
+	const totalContributors = contributors.length;
+
+	return (
+		<OpenSourceContributorsClient contributors={contributors} heading={heading} totalContributors={totalContributors} />
+	);
+}
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(sections)/contributors.tsx:25]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(sections)/open-issues.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+		<IssueClient
+			title={title}
+			issues={issues}
+			labels={labels}
+			tableHeader={tableHeader}
+			linkText={linkText}
+			filterText={filterText}
+		/>
+	);
+}
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(sections)/open-issues.tsx:37]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/FilterForm.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+			>
+				<option value={filterText}>{filterText}</option>
+				{labels.map((label) => (
+					<option key={label} value={label}>
+						{label}
+					</option>
+				))}
+			</select>
+		</section>
+	);
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/FilterForm.tsx:43]: fixed

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-commits.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+	if (linkHeader) {
+		const match = linkHeader.match(/&page=(\d+)>; rel="last"/);
+		if (match) {
+			totalCommits = parseInt(match[1], 10);
+		} else {
+			console.warn('Link header exists but could not parse total commits page count.');
+		}
+	} else {
+		console.warn('No Link header found; assuming a single commit.');
+	}
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-commits.tsx:85]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-forks.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+export async function getForkCount(): Promise<{ totalForks: number; newForks: number }> {
+	const repoUrl = `https://api.github.com/repos/${owner}/${repo}`;
+	const headers: Record<string, string> = {
+		Accept: 'application/vnd.github+json',
+	};
+	// Conditionally add the Authorization header if GITHUB_PAT is available
+	if (process.env.GITHUB_PAT) {
+		headers['Authorization'] = `Bearer ${process.env.GITHUB_PAT}`;
+	}
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-forks.tsx:17]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-forks.tsx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+		for (const fork of forks) {
+			if (new Date(fork.created_at) >= new Date(startDateISO)) {
+				newForks++;
+			}
+		}
+
+		// No more pages if we got fewer than 100 forks
+		if (forks.length < 100) hasMore = false;
+		page++;
+	}
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-forks.tsx:80]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-stars.tsx [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+			if (starredAt >= new Date(startDateISO)) {
+				newStars++;
+			}
+		}
+
+		// No more pages if we got fewer than 100 stars
+		if (stars.length < 100) hasMore = false;
+		page++;
+	}
+
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-stars.tsx:85]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-stars.tsx [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+		id: number;
+		login: string;
+	};
+	starred_at: string;
+}
+
+export async function getStarCount(): Promise<{ totalStars: number; newStars: number }> {
+	const repoUrl = `https://api.github.com/repos/${owner}/${repo}`;
+	const headers: Record<string, string> = {
+		Accept: 'application/vnd.github.star+json',
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-stars.tsx:15]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-contributors.tsx [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+	const res = await fetch(`https://api.github.com/repos/${owner}/${repo}/stats/contributors`, {
+		headers,
+	});
+
+	if (!res.ok) {
+		const errorDetails = await res.text();
+		const status = res.status;
+		if (status === 403) {
+			throw new Error(`GitHub API rate limit exceeded: ${status} - ${errorDetails}.`);
+		} else if (status === 404) {
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-contributors.tsx:37]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-contributors.tsx [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+	const headers: Record<string, string> = {
+		Accept: 'application/vnd.github+json',
+	};
+	// Conditionally add the Authorization header if GITHUB_PAT is available
+	if (process.env.GITHUB_PAT) {
+		headers['Authorization'] = `Bearer ${process.env.GITHUB_PAT}`;
+	}
+	const res = await fetch(`https://api.github.com/repos/${owner}/${repo}/stats/contributors`, {
+		headers,
+	});
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-contributors.tsx:30]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(sections)/overview.tsx [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+	const { totalStars, newStars } = await getStarCount();
+	const { newCommits, totalCommits } = await getCommits();
+	const { forks, commits, stars }: Overview = translator.t('overview');
+
+	return (
+		<section className="flex w-full flex-col flex-wrap gap-6 md:flex-row">
+			<OverviewCard
+				title={forks.title}
+				total={totalForks}
+				time={newForks > 0 ? `+${newForks} last month` : `${newForks} last month`}
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(sections)/overview.tsx:57]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(sections)/overview.tsx [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+
+export async function Overview({ lang }: DefaultParams) {
+	const translator = await Translator.getInstance({
+		language: lang,
+		namespaces: ['website-open-source'],
+	});
+
+	const { totalForks, newForks } = await getForkCount();
+	const { totalStars, newStars } = await getStarCount();
+	const { newCommits, totalCommits } = await getCommits();
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(sections)/overview.tsx:49]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(sections)/overview.tsx [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+				time={newCommits > 0 ? `+${newCommits} last month` : `${newCommits} last month`}
+			/>
+			<OverviewCard
+				title={stars.title}
+				total={totalStars}
+				time={newStars > 0 ? `+${newStars} last month` : `${newStars} last month`}
+			/>
+		</section>
+	);
+}
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(sections)/overview.tsx:71]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-issues.tsx [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+					}
+				});
+
+				return {
+					id: issue.id,
+					url: issue.html_url,
+					title: issue.title,
+					labels: issueLabel,
+				};
+			});
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-issues.tsx:72]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-issues.tsx [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+	};
+	if (process.env.GITHUB_PAT) {
+		headers['Authorization'] = `Bearer ${process.env.GITHUB_PAT}`;
+	}
+
+	const issues: Issue[] = [];
+	const labels: string[] = [];
+	let page = 1;
+	let hasMore = true;
+
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-issues.tsx:28]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/contributors-client.tsx [hunk 19]
```diff
@@ -0,0 +1,76 @@
+'use client';
+
+import { Avatar, AvatarFallback, AvatarImage, Button, Typography } from '@socialincome/ui';
+import { useState } from 'react';
+
+type ContributorProp = {
+	name: string;
+	commits: number;
+	avatarUrl: string;
+};
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/contributors-client.tsx:10]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/issues-client.tsx [hunk 20]
```diff
@@ -0,0 +1,80 @@
+'use client';
+
+import { Button, Table, TableBody, TableCell, TableHeader, TableRow, Typography } from '@socialincome/ui';
+import { useState } from 'react';
+import { FilterForm } from './FilterForm';
+import { Issue } from './issue';
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/issues-client.tsx:6]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-commits.tsx [hunk 21]
```diff
@@ -0,0 +1,91 @@
+const owner = 'socialincome-san';
+const repo = 'public';
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-commits.tsx:2]: not relevant

File: website/src/app/[lang]/[region]/(website)/open-source/(components)/get-commits.tsx [hunk 22]
```diff
@@ -0,0 +1,91 @@
...
+			);
+		} else if (status === 404) {
+			throw new Error(`GitHub repository ${owner}/${repo} not found.`);
+		} else {
+			throw new Error(`Failed to fetch recent commits from GitHub: ${status} - ${errorDetails}`);
+		}
+	}
+
+	const recentCommits: GitHubCommit[] = await res.json();
+
```
[the remark this answers is not in the document]
Comment (activus-d) [reply website/src/app/[lang]/[region]/(website)/open-source/(components)/get-commits.tsx:52]: not relevant

**Comment by activus-d** [comment]
> Nice! This looks good to me👍🏼
> 
> I made a small commit with a few cleanups, see :[20ca4c8](https://github.com/socialincome-san/public/commit/20ca4c8f1e937d0611e2c9f9b13d816fe6550a7b)
> 
> If you wanted, you could combine some of the logic from the `get-commits.ts`, `get-forks.ts`, etc. files because they have quite a bit of overlapping functionality. Otherwise, it's all good😊


Thank you for the feedback, @mkue. I'll add the logic combination to my tasks. For learning purposes, I'm curious about why you changed the file format of some fetching files from `.tsx` to `.ts`.

**Comment by mkue** [comment]
> > Nice! This looks good to me👍🏼
> > I made a small commit with a few cleanups, see :[20ca4c8](https://github.com/socialincome-san/public/commit/20ca4c8f1e937d0611e2c9f9b13d816fe6550a7b)
> > If you wanted, you could combine some of the logic from the `get-commits.ts`, `get-forks.ts`, etc. files because they have quite a bit of overlapping functionality. Otherwise, it's all good😊
> 
> Thank you for the feedback, @mkue. I'll add the logic combination to my tasks. For learning purposes, I'm curious about why you changed the file format of some fetching files from `.tsx` to `.ts`.

The `.tsx` file extension is for files that contain JSX syntax (the html tags, see https://react.dev/learn/writing-markup-with-jsx). If a file does not contain any of these tags, it's a regular TypeScript file and should have the `.ts` file extension.

**Pull request merged**